packages/forms/src/UIForm/UIForm.container.js

```diff
@@ -5,28 +5,87 @@ import React from 'react';
 import UIFormTranslatedComponent from './UIForm.component';
 import { formPropTypes } from './utils/propTypes';
 
+/**
+ * add error object on a formSchema if it doesn't exist
+ * @param {FormSchema} formSchema
+ * @return {FormSchema}
+ */
+function addErrorObject(formSchema) {
+	if (!formSchema.errors) {
+		return { errors: {}, ...formSchema };
+	}
+	return formSchema;
+}
+
+/**
+ * reinit liveState with empty errors if it doesn't exist
+ * @param {FormSchema} newFormSchema
+ * @param {State} prevState
+ * @return {State}
+ */
+const reinitLiveState = newFormSchema => () => ({
+	liveState: addErrorObject(newFormSchema),
+});
+
+/**
+ * update live state with new properties derived from user interacting
+ * with the form
+ * @param {Object} properties
+ * @param {State} prevState
+ * @return {State}
+ */
+const change = properties => prevState => ({
+	...prevState,
+	liveState: { ...prevState.liveState, properties },
+});
+
+/**
+ * update liveState formSchema with errors
+ * @param {Object} errors
+ * @param {State} prevState
+ * @return {State}
+ */
+const setErrors = errors => prevState => ({
+	...prevState,
+	liveState: { ...prevState.liveState, errors },
+});
+
+/**
+ * update initialState with liveState after a user submission of the form
+ * @param {Object} newProperties
+ * @param {State} prevState
+ * @return {State}
+ */
+const submit = newProperties => prevState => {
+	const newFormSchema = { ...prevState.liveState, properties: newProperties };
```

Comment:
You don't need the replace the properties / create a new object here.
On submit, the submitted properties are the current live form properties.